docs/DesignDocs/RemoteAttestationCollaterals.md

```diff
@@ -0,0 +1,292 @@
+Remote Attestation Collaterals
+=====
+
+Adding attestation APIs to support collaterals for remote attestation.  Collaterals are information from a second source that the user/challenger can provide for attestation verification.
+
+Motivation
+----------
+
+Currently, the existing collaterals used for Intel SGX quote verification are not exposed to the user.  This makes it difficult for the user/challenger to specify his/her own set of policies.  Adding these new APIs allows the user to specify a validation policy of his/her choosing.  Possible policies:
+1. The user/challenger has the option to specify its set of collaterals during verification.
+2. The user/challenger has the option to provide a datetime to use during verification.  This datetime specified the date and time at which the user wants to do the verification.  If no datetime is provided, it uses the datetime when the collaterals were created.  The user can provide a datetime in the past, enabling auditing of the evidence and collaterals.
+
+
+User Experience
+---------------
+
+There are 3 scenarios.
+
+### 1. Challenger does not specify the collaterals:
+In this scenario the challenger does not receive any collaterals from the attester(where the TEE resides).  The challenger also does not pass any collaterals to the verification function, instead the collaterals are fetched during verification by the OE SDK.
```

Comment:
As phrased I think the first sentence assumes the challenger is the Verifier not the Relying Party, since a Relying Party would receive info from a Verifier instead of an Attester.  Am I interpreting the intent correctly?

Reply:
Yes.